scala/lambdas/ingest-flow-control/src/test/scala/uk/gov/nationalarchives/ingestflowcontrol/Helpers.scala

```diff
@@ -89,15 +89,19 @@ object Helpers {
     override def writeItems[T](tableName: String, items: List[T])(using format: DynamoFormat[T]): IO[List[BatchWriteItemResponse]] = notImplemented
 
     override def queryItems[U](tableName: String, requestCondition: RequestCondition, potentialGsiName: Option[String])(using returnTypeFormat: DynamoFormat[U]): IO[List[U]] =
-      notImplemented
-
-    override def getItems[T, K](primaryKeys: List[K], tableName: String)(using returnFormat: DynamoFormat[T], keyFormat: DynamoFormat[K]): IO[List[T]] =
-      val firstPK = primaryKeys.head.asInstanceOf[IngestQueuePartitionKey]
-      errors.raise(_.getItem, "Error getting item from dynamo table") >>
+      errors.raise(_.queryItem, "Error getting item from dynamo table") >>
         ref.get.map { existing =>
-          existing.filter(_.sourceSystem == firstPK.sourceSystem).sortBy(_.queuedAt).map(_.asInstanceOf[T])
+          (for {
+            values <- Option(requestCondition.attributes.values)
+            map <- values.toMap[String].toOption
+          } yield existing
+            .filter(row => map.get("conditionAttributeValue0").contains(row.sourceSystem))
```

Comment:
1. Should we use "item" rather than "row"?
2. Is `conditionAttributeValue0` an real value on an item?

Reply:
I think its a placeholder